src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -563,17 +565,21 @@ void c4StatusChanged(C4ReplicatorStatus c4Status) {
                 if (h != null) { responseHeaders = FLValue.fromData(h).asDict(); }
             }
 
-            if (c4Status.getActivityLevel() > C4ReplicatorStatus.ActivityLevel.CONNECTING) {
-                resetRetryCount();
-                if (reachabilityManager != null) { reachabilityManager.removeNetworkReachabilityListener(this); }
-            }
-            else if (c4Status.getActivityLevel() == C4ReplicatorStatus.ActivityLevel.STOPPED) {
-                final C4Error error = c4Status.getC4Error();
-                if (actuallyMeansOffline(error)) {
-                    handleError(error);
-                    // Change c4Status to offline, so my state will reflect that, and proceed:
-                    c4Status.setActivityLevel(C4ReplicatorStatus.ActivityLevel.OFFLINE);
-                }
+            switch (c4ActivityLevel) {
```

Comment:
this is same as the old code, but changed the if..else to a switch. right?

Reply:
Yeah.  I've stumbled over that `>` in the first comparison, twice now.  I want to make it clear what the condition is.